tests/storage/test_storage.py

```diff
@@ -621,6 +621,30 @@ def test_download_file(self):
         finally:
             os.remove(file.name)
 
+    def test_copy_file(self):
+        # Store a file in a different source bucket
+        source_store = storage.MemoryFileStore("source-bucket")
+        contents = b"test_download_file"
+        source_bucket_name, source_key_path = source_store.store_file(
+            namespace="mem",
+            filename="test.pdf",
+            contents=contents,
+        )
+        s3_object = storage.S3Object(
+            bucket_name=source_bucket_name,
+            key=source_key_path,
+        )
+        destination = "a/b/c.pdf"
```

Comment:
🐼 Since the original key path is called `source_key_path`, shall we rename this as `destination_key_path`?